apps/meeting/src/providers/VideoGridProvider/state.tsx

```diff
@@ -0,0 +1,510 @@
+import {
+  DefaultModality,
+  TargetDisplaySize,
+  VideoPreference,
+  VideoPreferences,
+  VideoSource,
+} from 'amazon-chime-sdk-js';
+import { priorityBasedPolicy } from '../../meetingConfig';
+import { VideoGridMode } from '../../types';
+
+interface GridState {
+  mode: VideoGridMode;
+  isZoomed: boolean;
+  threshold: number;
+}
+
+interface AttendeeState {
+  attendeeId: string;
+  name: string;
+  videoEnabled: boolean;
+  bandwidthConstrained: boolean;
+}
+interface VideoSourceState {
+  cameraSources: string[];
+  activeSpeakersWithCameraSource: string[];
+  contentShareId: string | null;
+  hasLocalVideo: boolean;
+  hasLocalContentSharing: boolean;
+}
+
+type VideoSourceWithType = { attendeeId: string; type: string };
+
+export type State = {
+  gridState: GridState;
+  attendeeStates: { [attendeeId: string]: AttendeeState };
+  videoSourceState: VideoSourceState;
+};
+
+export const initialState: State = {
+  gridState: {
+    mode: VideoGridMode.GalleryView,
+    isZoomed: false,
+    threshold: 8,
+  },
+  attendeeStates: {},
```

Comment:
Suggest changing `attendeeStates` to just `attendees`, or use `attendeeState` while looping over these down below in the reducer actions.